config/initializers/carrierwave.rb

```diff
@@ -6,40 +6,43 @@
   config.timeout = 10
 end
 
-# rubocop:disable Metrics/BlockLength
-CarrierWave.configure do |config|
-  if Rails.env.test? || Rails.env.development?
-    config.storage = :file
-    config.enable_processing = Rails.env.development?
-  elsif ENV["FILE_STORAGE_LOCATION"] == "file" # @forem/systems production version of file store
-    config.asset_host = "https://#{ApplicationConfig['APP_DOMAIN']}/localimages"
-    config.storage = :file
-  else
-    config.fog_provider = "fog/aws"
-    if ENV["FOREM_CONTEXT"] == "forem_cloud" # @forem/systems jdoss's special sauce.
-      config.fog_credentials = {
-        provider: "AWS",
-        use_iam_profile: true,
-        region: "us-east-2"
-      }
-      config.asset_host = "https://#{ApplicationConfig['APP_DOMAIN']}/remoteimages"
-      config.fog_public = false
-    elsif ApplicationConfig["AWS_ID"].present?
-      region = ApplicationConfig["AWS_UPLOAD_REGION"].presence || ApplicationConfig["AWS_DEFAULT_REGION"]
-      config.fog_credentials = {
-        provider: "AWS",
-        aws_access_key_id: ApplicationConfig["AWS_ID"],
-        aws_secret_access_key: ApplicationConfig["AWS_SECRET"],
-        region: region
-      }
-      config.fog_attributes = { cache_control: "public, max-age=#{365.days.to_i}" }
-    else
-      # Fallback on file storage if AWS creds are not present
-      config.asset_host = "https://#{ApplicationConfig['APP_DOMAIN']}/localimages"
-      config.storage = :file
-    end
+disk_storage = proc do |config|
+  config.storage = :file
+  config.enable_processing = !Rails.env.test?
+  config.asset_host = if Rails.env.production?
+                        "https://#{ApplicationConfig['APP_DOMAIN']}/localimages"
+                      end
+end
+
+if !Rails.env.production? || ENV["FILE_STORAGE_LOCATION"] == "file"
+  CarrierWave.configure(&disk_storage)
```

Comment:
Is there a specific reason for `disk_storage` to be a proc while the other branches of this `if` are inlined?

Reply:
I know the purpose of the proc is to de-duplicate code but I think it would read a little cleaner without it.

Reply:
Yeah, I wasn't really happy with this refactor either. I just re-arranged it differently and removed this proc.